src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -1916,7 +1916,8 @@ void MergeTreeData::clearPartsFromFilesystemImpl(const DataPartsVector & parts_t
     const auto settings = getSettings();
     if (parts_to_remove.size() > 1
         && settings->max_part_removal_threads > 1
-        && parts_to_remove.size() > settings->concurrent_part_removal_threshold)
+        && parts_to_remove.size() > settings->concurrent_part_removal_threshold
+        && (!supportsReplication() || !settings->allow_remote_fs_zero_copy_replication)) /// parts must be removed in order for zero-copy replication
```

Comment:
The comment does not explain "why".

It reads like this:
```
/// parts must be removed in order for zero-copy replication
```
what the heck is "zero-copy" replication and why it requires part removal in order? Is it a kludge and is the code of "zero-copy" replication trashy?